framework/src/Volo.Abp.EventBus/Volo/Abp/EventBus/EventHandlerInvoker.cs

```diff
@@ -9,18 +10,26 @@ namespace Volo.Abp.EventBus;
 public class EventHandlerInvoker : IEventHandlerInvoker, ISingletonDependency
 {
     private const string EventHandlerMethodName = "HandleEventAsync";
-    private readonly ConcurrentDictionary<Type, EventHandlerMethodExecutor> _executorCache;
+    private readonly ConcurrentDictionary<string, EventHandlerMethodExecutor> _executorCache;
 
     public EventHandlerInvoker()
     {
-        _executorCache = new ConcurrentDictionary<Type, EventHandlerMethodExecutor>();
+        _executorCache = new ConcurrentDictionary<string, EventHandlerMethodExecutor>();
     }
 
-    public Task InvokeAsync(IEventHandler eventHandler, object eventData)
+    public Task InvokeAsync(IEventHandler eventHandler, object eventData, Type eventType)
     {
         var handleType = eventHandler.GetType();
-        var executor = _executorCache.GetOrAdd(handleType, type => EventHandlerMethodExecutor.Create(type.GetMethod(EventHandlerMethodName), type.GetTypeInfo()));
+        var key = $"{handleType.FullName}_{eventType.FullName}";
+
+        var executor = _executorCache.GetOrAdd(key, _ => EventHandlerMethodExecutor.Create(GetHandleEventMethodInfo(handleType, eventType), handleType.GetTypeInfo()));
 
         return executor.ExecuteAsync(eventHandler, new[] { eventData });
     }
+
+    private static MethodInfo GetHandleEventMethodInfo(Type handleType, Type eventType)
+    {
+        var methods = handleType.GetMethods().Where(x => x.Name == EventHandlerMethodName).ToArray();
+        return methods.Length == 1 ? methods.First() : methods.FirstOrDefault(x => x.GetParameters().Any(param => param.ParameterType == eventType));
```

Comment:
I'm not sure but I think in inheritance this could select wrong method to execute.
Like this use-case:
```
public class BaseClass{}
public class Foo : BaseClass 
{
}
public class Bar: BaseClass
{
}

public class FooBarEventHandler : ILocalEventHandler<BaseClass>
{
    public Task HandleEventAsync(BaseClass eventData)
    {
        Dosomthing();
        if(eventData is Foo foo)
        {
             HandleEventAsync(foo);
        }
    }

    private Task HandleEventAsync(Foo eventData)
    {
         return Task.CompletedTask;
    }
}
```
This would directly execute `HandleEventAsync(Foo eventData)` if the `eventType` be `Foo`  while it should execute ` HandleEventAsync(BaseClass eventData)`

Reply:
Hi, 
Did you try to test it?

Use `handlerFactories.EventType` instead of the `parameter type` when triggering handlers, it will work as expected

https://github.com/abpframework/abp/blob/e3e1779de6df5d26f01cdc8e99ac9cbcb3d24d3c/framework/src/Volo.Abp.EventBus/Volo/Abp/EventBus/EventBusBase.cs#L139

You can check this unit test

https://github.com/abpframework/abp/blob/e3e1779de6df5d26f01cdc8e99ac9cbcb3d24d3c/framework/test/Volo.Abp.EventBus.Tests/Volo/Abp/EventBus/Local/EventBus_MultipleHandle_Test.cs#L12

Reply:
Sorry my mistake, I was triggering with `eventType` instead of `handlerFactories.EventType`